### DISCUSSION TITLE
Sphinx showing tuple doc strings in using qcodes 0.32.0

### DISCUSSION THREAD

**Discussion by ereehuis** [microsoft/Qcodes#3955] [Q&A] [answered]
Hi! I'm chasing an elusive bug in building our `quantify-core` documentation

=> In using `qcodes` **0.32.0**, I get doc strings from `tuple` built-in instead of our own doc strings 

<img src="https://user-images.githubusercontent.com/35330766/155132757-9dea5542-dc35-47d9-8850-cba476c922af.png" width="70%" />

```
python -m pydoc tuple

Help on class tuple in module builtins:

class tuple(object)
 |  tuple(iterable=(), /)
 |
 |  Built-in immutable sequence.
 |
 |  If no argument is given, the constructor returns an empty tuple.
 |  If iterable is specified the tuple is initialized from iterable's items.
 |
 |  If the argument is a tuple, the return value is the same object.
```

=> When changing to `qcodes` **0.31.0**, all is fine

<img src="https://user-images.githubusercontent.com/35330766/155130217-297fe924-28a9-4f3a-9499-2c68fba43c96.png" width="65%" />

=> Lastly, when commenting out `"qcodes.sphinx_extensions.parse_parameter_attr"` in our sphinx extensions  https://gitlab.com/quantify-os/quantify-core/blob/c33b26d589672a6e2598b074d39d70afc678ba17/docs/conf.py#L72-L72
methods that showed the `tuple` doc string before are now empty, as a second confirmation of the link to `qcodes`  / `qcodes.sphinx_extensions.parse_parameter_attr`

<img src="https://user-images.githubusercontent.com/35330766/155132417-01135baa-7901-462e-84da-89c59d79a3a8.png" width="45%" />

=> Have you ever seen this `tuple` message show up in docs like this, any idea how to circumvent or what change/PR might have led to this? 🙏 

Scanned through PRs merged into version 0.32.0, found this one that looked interesting as it deals with doc strings and Tuple and instruments, but is a very loose lead: https://github.com/QCoDeS/Qcodes/pull/3864/files

**Comment by jenshnielsen** [comment, marked as answer]
Thats strange. I don't see any obvious reason for why this is happening. 
We are not seeing any issue like this in the qcodes documentation. 

Could you try to bisect between 0.31 and 0.32 to see which commit triggered this? 
https://www.tutorialdocs.com/article/git-bisect-tutorial.html

**Reply by ereehuis** [reply]
Will do, thanks for the suggestion 👍

**Reply by ereehuis** [reply]
Actually, issue with Ellipsis in docstrings was due to us using Sphinx v3. I'm moving us to Sphinx v4 which resolves the issue (see https://gitlab.com/quantify-os/quantify-core/-/merge_requests/317)

Thanks again!

**Reply by jenshnielsen** [reply]
Thanks for the info. We don't put a min supported sphinx version in our docs but perhaps we should
